apps/barista-design-system/src/shared/services/scroll-spy.service.ts

```diff
@@ -14,81 +14,113 @@
+      .subscribe(() => {
+        this._onResize();
+      });
+    fromEvent(window, 'scroll')
+      .pipe(auditTime(10), takeUntil(this._onStopListening))
+      .subscribe(() => {
+        this._onScrollEvent();
+      });
+    this._onResize();
+    this._spiedElements.forEach((_) => {
+      this._onScroll(this._getScrollTop(), this._lastMaxScrollTop);
+    });
+
+    return this.activeScrollItem.asObservable().pipe(distinctUntilChanged());
   }
-}
 
-/*
- * Represents a group of "scroll-spied" elements. Contains info and methods for efficiently
- * determining which element should be considered "active", i.e. which element has been scrolled
- * passed the top of the viewport.
- */
-export class BaScrollSpiedElementGroup {
-  activeScrollItem: ReplaySubject<BaScrollItem | null> = new ReplaySubject(1);
-  private _spiedElements: BaScrollSpiedElement[];
+  /**
+   * Stop tracking this group of elements and emitting active elements. If there is no other group
+   * being spied, stop listening for `resize` or `scroll` events.
+   */
+  unspy(): void {
+    this.activeScrollItem.complete();
+    this._spiedElements = [];
+    if (!this._spiedElements.length) {
+      this._onStopListening.next();
+    }
+  }
 
-  constructor(elements: Element[]) {
-    this._spiedElements = elements.map(
-      (elem, i) => new BaScrollSpiedElement(elem, i),
-    );
+  /**
+   * @internal The size of the window has changed. Re-calculate all affected values,
+   * so that active elements can be determined efficiently on scroll.
+   */
+  private _onResize(): void {
+    this._lastContentHeight = this._getContentHeight();
+    this._lastMaxScrollTop =
+      this._getContentHeight() - this._getViewportHeight();
+    this._calculateTopOfSpyElements();
   }
 
-  /*
-   * Calculate the `top` value of each ScrollSpiedElement of this group (based on te current
-   * `scrollTop` and `topOffset` values), so that the active element can be later determined just by
-   * comparing its `top` property with the then current `scrollTop`.
+  /**
+   * @internal Determine which element for each ScrollSpiedElementGroup is active. If the content height has
+   * changed since last check, re-calculate all affected values first.
    */
-  calibrate(scrollTop: number, topOffset: number): void {
-    for (const spiedElem of this._spiedElements) {
-      spiedElem.calculateTop(scrollTop, topOffset);
+  private _onScrollEvent(): void {
+    if (this._lastContentHeight !== this._getContentHeight()) {
+      this._onResize();
     }
-    this._spiedElements.sort((a, b) => b.top - a.top); // Sort in descending `top` order.
+
+    this._spiedElements.forEach((_) => {
+      this._onScroll(this._getScrollTop(), this._lastMaxScrollTop);
+    });
   }
 
-  /*
-   * Determine which element is the currently active one, i.e. the lower-most element that is
+  /**
+   * @internal Determine which element is the currently active one, i.e. the lower-most element that is
```

Comment:
No need for internal if it is private.
```suggestion
   * Determine which element is the currently active one, i.e. the lower-most element that is
```